lib/streamlit/web/cli.py

```diff
@@ -53,10 +53,19 @@ def _convert_config_option_to_click_option(
             f"\n {config_option.deprecation_text} - {config_option.expiration_date}"
         )
 
+    # Special handling for headingFontWeights which needs JSON parsing
+    # Use str type for CLI to avoid Click splitting JSON into individual characters
+    cli_type = config_option.type
+    if config_option.key in (
+        "theme.headingFontWeights",
+        "theme.sidebar.headingFontWeights",
+    ):
+        cli_type = str
```

Comment:
nit: is there a benefit of specifying `list[int]` for `headingFontWeights` as type vs. just leaving it as a generic string? I think a couple of other lists like `chartCategoricalColors` is also just treated as a str to allow json list input instead of the repeated cli args pattern that is usually chosen by click for lists.